cmake/modules/BuildRocksDB.cmake

```diff
@@ -86,6 +93,15 @@ function(build_rocksdb)
 
   add_library(RocksDB::RocksDB STATIC IMPORTED)
   add_dependencies(RocksDB::RocksDB rocksdb_ext)
+  if(WITH_LIBURING)
+    # None of these work!
+    #add_dependencies(RocksDB::RocksDB INTERFACE uring::uring)
+    #add_dependencies(rocksdb_ext uring::uring)
+    #target_link_libraries(rocksdb_ext PRIVATE uring::uring)
+    #target_link_libraries(RocksDB::RocksDB PRIVATE uring::uring)
+    #target_link_libraries(RocksDB::RocksDB INTERFACE uring::uring)
+    #add_dependencies(RocksDB::RocksDB uring::uring)
```

Comment:
Unfortunately, none of these commands work to correctly order the dependency of `src/rocksdb/librocksdb.a` on `src/liburing/src/liburing.a`. I've done a lot of googling and experimentation without success to get this to work.

Reply:
cc @cbodley @rzarzynski 

Any other cmake experts that may have an idea?

Reply:
other rocksdb dependencies are added above with the form:
```cmake
  if(SNAPPY_FOUND)
    list(APPEND rocksdb_INTERFACE_LINK_LIBRARIES snappy::snappy)
  endif()
```
then applied at the end:
```cmake
  set_target_properties(RocksDB::RocksDB PROPERTIES
    INTERFACE_INCLUDE_DIRECTORIES "${rocksdb_INCLUDE_DIR}"
    INTERFACE_LINK_LIBRARIES "${rocksdb_INTERFACE_LINK_LIBRARIES}"
```
i'd expect that use of `INTERFACE_LINK_LIBRARIES` to be equivalent to your line:
```cmake
target_link_libraries(RocksDB::RocksDB INTERFACE uring::uring)
```
is it possible that `set_target_properties()` is overwriting that?

Reply:
> other rocksdb dependencies are added above with the form:
> 
> ```cmake
>   if(SNAPPY_FOUND)
>     list(APPEND rocksdb_INTERFACE_LINK_LIBRARIES snappy::snappy)
>   endif()
> ```

I can't believe I missed this. That was all we needed!

Reply:
sorry, i missed this. i started commenting right after reading "correctly order the dependency of".

> list(APPEND rocksdb_CMAKE_ARGS -DWITH_GFLAGS=OFF)

could you help me understand how this is related to liburing?

also, i'd suggest to split this change into two. as they addresses two different issues:

1. promote the uring to top source directory
2. populate liburing include and library paths down to rocksdb external project

Reply:
I don't recall why I turned GFLAGS off. I reverted that and applied your other suggestions. Thanks @tchaikov